pkg/registry/apps/deployment/storage/storage.go

```diff
@@ -329,6 +330,18 @@ func scaleFromDeployment(deployment *apps.Deployment) (*autoscaling.Scale, error
 	if err != nil {
 		return nil, err
 	}
+
+	// The scale managed fields must contain all the managers of the deployment
+	// that owns .spec.replicas
+	managedFieldsEntries, err := fieldmanager.FindEntries(deployment.ManagedFields, "spec", "replicas")
+	if err != nil {
+		return nil, err
+	}
+	for i := range managedFieldsEntries {
+		managedFieldsEntries[i].APIVersion = "autoscaling/v1"
+		managedFieldsEntries[i].FieldsV1 = &metav1.FieldsV1{Raw: []byte(`{"f:spec":{"f:replicas":{}}}`)}
```

Comment:
Building the field manually is quite unfortunate.

Reply:
Also, I think we agreed for now that this would always be a constant final path (since that's an autoscaling field). Do we need to have the logic here where it would have to be repeated?